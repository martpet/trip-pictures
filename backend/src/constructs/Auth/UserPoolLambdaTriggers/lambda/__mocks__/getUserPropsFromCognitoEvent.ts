export const getUserPropsFromCognitoEvent = vi
  .fn()
  .mockName('getUserPropsFromCognitoEvent')
  .mockReturnValue({
    id: 'dummyId',
    dummyUserPropFromCognitoEventKey: 'dummyUserPropFromCognitoEventValue',
  });
