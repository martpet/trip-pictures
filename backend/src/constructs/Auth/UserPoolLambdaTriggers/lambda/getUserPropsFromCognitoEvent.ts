import {
  CognitoIdentity,
  IdentityProvider,
  PostAuthenticationTriggerEvent,
  PostConfirmationTriggerEvent,
  UserPropsFromCognito,
} from 'lambda-layer';

type TriggerEvent = PostConfirmationTriggerEvent | PostAuthenticationTriggerEvent;

export const getUserPropsFromCognitoEvent = (
  event: TriggerEvent
): UserPropsFromCognito => {
  const { userAttributes } = event.request;
  const identity = JSON.parse(userAttributes.identities)[0] as CognitoIdentity;

  return {
    id: userAttributes.sub,
    providerName: identity.providerName as IdentityProvider,
    givenName: userAttributes.given_name,
    familyName: userAttributes.family_name,
    picture: userAttributes.picture as string | undefined,
    email: userAttributes.email,
    dateCreated: identity.dateCreated,
  };
};
