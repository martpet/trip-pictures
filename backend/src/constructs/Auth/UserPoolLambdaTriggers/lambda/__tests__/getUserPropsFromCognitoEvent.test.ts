import { getUserPropsFromCognitoEvent } from '../getUserPropsFromCognitoEvent';
import event from '../postConfirmation/__tests__/__fixtures__/postConfirmationEvent';

describe('getUserPropsFromCognitoEvent', () => {
  it('returns a correct value', () => {
    expect(getUserPropsFromCognitoEvent(event)).toMatchSnapshot();
  });
});
