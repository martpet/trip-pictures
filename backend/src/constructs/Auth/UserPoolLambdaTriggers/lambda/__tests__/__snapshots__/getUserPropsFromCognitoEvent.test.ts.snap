// Vitest Snapshot v1

exports[`getUserPropsFromCognitoEvent > returns a correct value 1`] = `
{
  "dateCreated": 1658604803064,
  "email": "dummyPostConfirmEmail",
  "familyName": "dummyPostConfirmFamilyName",
  "givenName": "dummuPostConfirmGivenName",
  "id": "dummyPostConfirmSub",
  "picture": "dummyPostConfirmPictureUrl",
  "providerName": "Google",
}
`;
